src/nupic/algorithms/Connections.cpp

```diff
@@ -340,6 +445,68 @@ UInt Connections::numSynapses() const
   return numSynapses_;
 }
 
+bool Connections::operator== (const Connections &other) const
```

Comment:
I don't think that is a thing in C++, `operator==` seems right to me (nit: no space before open paren)